tchannel/schemes/thrift.py

```diff
@@ -58,6 +59,15 @@ def __call__(self, request=None, headers=None, timeout=None,
 
         # success - non-void
         if len(result_spec) >= 1 and result_spec[0] is not None:
+
+            # value expected, but got none
+            # TODO - server side should use this same logic
+            if body.success is None:
+                raise ValueExpectedError(
+                    'called a non-void procedure %s but received no value' %
```

Comment:
Actually... I like somehow indicating that non-void procedures need to return values... or that only void procedures can return None